radio/src/gui/colorlcd/radio_theme.cpp

```diff
@@ -0,0 +1,723 @@
+        new ThemeDetailsDialog(page, _theme, [=] (ThemeFile t) {
+          _theme.setAuthor(t.getAuthor());
+          _theme.setInfo(t.getInfo());
+          _theme.setName(t.getName());
+
+          // update the theme name
+          themeName->setText(_theme.getName());
+        });
+        return 0;
+      }, BUTTON_BACKGROUND | OPAQUE, textFont);
+
+      // setup the prev next controls so save and details are in the mix
+      window->link(cList, detailButton);
+      window->link(saveButton, cList);
+    }
+
+    void buildBody(FormGroup *window)
+    {
+      rect_t r = { 0, 4, COLOR_LIST_WIDTH, COLOR_LIST_HEIGHT};
+      cList = new ColorList(window, r, _theme.getColorList());
+      cList->setLongPressHandler([=] (event_t event) {
+        auto colorEntry = cList->getSelectedColor();
+        new ColorEditPage(_theme, colorEntry.colorNumber, 
+        [=] (ThemeFile *theme) {
+          if (previewWindow != nullptr && cList != nullptr) {
+            _theme.setColor(colorEntry.colorNumber, 
+                            theme->getColorEntryByIndex(colorEntry.colorNumber)->colorValue);
+            cList->setColorList(_theme.getColorList());
+            colorMaintainer.setColorList(_theme.getColorList());
+            previewWindow->invalidate();
+          }
+        });
+      });
+
+      if (LCD_W > LCD_H) {
+        r = { COLOR_LIST_WIDTH + MARGIN_WIDTH, 4, LCD_W - COLOR_LIST_WIDTH - MARGIN_WIDTH*2, COLOR_LIST_HEIGHT };
+      } else {
+        r = { 0, LEFT_LIST_HEIGHT + 4,  LEFT_LIST_WIDTH, LEFT_LIST_HEIGHT - 4};
+      }
+      previewWindow = new PreviewWindow(window, r, _theme.getColorList());
+    }
+
+  protected:
+    ThemeFile _theme;
+    Page *page;
+    std::function<void(ThemeFile &theme)> saveHandler = nullptr;
+    PreviewWindow *previewWindow = nullptr;
+    ColorList *cList = nullptr;
+    StaticText *themeName = nullptr;
+    TextButton *saveButton;
+    TextButton *detailButton;
+};
+
+ThemeSetupPage::ThemeSetupPage() : PageTab(STR_THEME_EDITOR, ICON_RADIO_EDIT_THEME) {}
+ThemeSetupPage::~ThemeSetupPage()
+{
+}
+
+void ThemeSetupPage::setupListbox(FormWindow *window, rect_t r, ThemePersistance *tp)
+{
+  listBox = new ListBox(
+    window, r, tp->getNames(),
+      [=]() { 
+        return currentTheme;
+      },
+      [=](uint8_t value) {
+        if (themeColorPreview && authorText && nameText && fileCarosell) {
+          ThemeFile *theme = tp->getThemeByIndex(value);
+          themeColorPreview->setColorList(theme->getColorList());
+          authorText->setText(theme->getAuthor());
+          nameText->setText(theme->getName());
+          fileCarosell->setFileNames(theme->getThemeImageFileNames());
+        }
+        currentTheme = value;
+      });
+  listBox->setActiveIndex(tp->getThemeIndex());
+  listBox->setTitle(STR_THEME + std::string("s"));
+  listBox->setLongPressHandler([=] (event_t event) {
+    auto menu = new Menu(listBox,false);
+
```

Comment:
How about adding adding `listBox->getSelected() != tp->getThemeIndex()` check to hide STR_ACTIVATE line if trying to make active theme active?